routes/helpers/firewall.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Clover
+  def firewall_list_dataset
+    @project.firewalls_dataset.authorized(current_account.id, "Firewall:view")
+  end
+
+  def firewall_list_api_response(dataset)
+    dataset = dataset.where(location: @location) if @location
+    result = dataset.eager(:firewall_rules).paginated_result(
+      start_after: request.params["start_after"],
+      page_size: request.params["page_size"],
+      order_column: request.params["order_column"]
+    )
+
+    {
+      items: Serializers::Firewall.serialize(result[:records]),
+      count: result[:count]
+    }
+  end
+
+  def firewall_post(firewall_name)
+    Authorization.authorize(current_account.id, "Firewall:create", @project.id)
+    Validation.validate_name(firewall_name)
+    @location ||= LocationNameConverter.to_internal_name(request.params["location"])
+
+    description = if api?
+      Validation.validate_request_body(request.body.read, [], ["description"])["description"] || ""
+    else
+      request.params["description"]
+    end
```

Comment:
I think we can merge them using json_params similar to vms, private subnets
https://github.com/ubicloud/ubicloud/blob/e80bd6e062b6474a707774d7a33cd50e0db4352a/routes/helpers/vm.rb#L27-L31